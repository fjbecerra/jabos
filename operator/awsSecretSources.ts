export default function (aws: { secret: string, accessKeyId: string, secretAccessKey: string }): any[] {
  return !aws ? [] : [
    {
      "secret": {
        "name": aws.secret,
        "items": [
          {
            "key": aws.accessKeyId,
            "path": "aws_access_key_id"
          },
          {
            "key": aws.secretAccessKey,
            "path": "aws_secret_access_key"
          }
        ]
      }
    }
  ]
}
