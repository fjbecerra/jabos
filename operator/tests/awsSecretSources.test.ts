import awsSecretSources from '../awsSecretSources';
import { clearMock, setMock } from './settingsMock';

beforeEach(() => {
  setMock();
});

afterEach(() => {
  clearMock();
});

test('awsSecretSources without creds', () => {
  expect(awsSecretSources(undefined)).toMatchSnapshot();
});

test('awsSecretSources with creds', () => {
  const awsCreds = {
    secret: 'secret_name',
    accessKeyId: 'accessKeyId_name',
    secretAccessKey: 'secretAccessKey_name'
  }
  expect(awsSecretSources(awsCreds)).toMatchSnapshot();
});
