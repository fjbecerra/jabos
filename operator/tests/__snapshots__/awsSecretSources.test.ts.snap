// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`awsSecretSources with creds 1`] = `
Array [
  Object {
    "secret": Object {
      "items": Array [
        Object {
          "key": "accessKeyId_name",
          "path": "aws_access_key_id",
        },
        Object {
          "key": "secretAccessKey_name",
          "path": "aws_secret_access_key",
        },
      ],
      "name": "secret_name",
    },
  },
]
`;

exports[`awsSecretSources without creds 1`] = `Array []`;
